src/main.rs

```diff
@@ -0,0 +1,262 @@
+            Ok(())
+        } else {
+            Err("Project not found".into())
+        }
+    } else if matches.is_present("stop") {
+        let res = t.get_running_entry().expect("API Error");
+        if let Some(current_entry) = res {
+            t.stop_entry(&current_entry).expect("Error");
+            Ok(())
+        } else {
+            Err("No task currently running".into())
+        }
+    } else if matches.is_present("swap") {
+        let mut entries = get_todays_stored_entries(t);
+        if entries.is_empty() {
+            return Err("Not enough entries stored to swap".into());
+        }
+
+        entries.sort_by(|a, b| b.cmp(a)); //reverse it
+        t.start_entry(&entries[0].description, &[], &entries[0].project)
+            .expect("API Error");
+        Ok(())
+    } else if let Some(id_string) = matches.value_of("delete") {
+        let entries = get_todays_stored_entries(t);
+        let id = id_string.parse::<usize>();
+        if let Ok(id) = id {
+            println!("len, id {} {}", entries.len(), id);
+            if id - 1 < entries.len() {
+                t.delete_entry(&entries[id - 1]).expect("API Error");
+                Ok(())
+            } else {
+                Err("You tried to delete and entry that does not exist".into())
+            }
+        } else {
+            Err("Could not parse id".into())
+        }
+    } else if let Some(mut new) = matches.values_of("edit") {
+        let id: Option<usize> = new.next().and_then(|s| s.parse::<usize>().ok());
+        let new_description = new.next();
+        let new_project = new.next().and_then(|s| s.parse::<usize>().ok());
+
+        let entries = get_todays_stored_entries(t);
+        if entries.is_empty() {
+            Err("Not enough entries stored to edit".into())
+        } else if let Some(id) = id {
+            if (id - 1 < entries.len())
+                & (new_description.is_some())
+                & (new_project.is_some())
+                & (new_project.unwrap_or(0) < projects.len())
+            {
+                let project = projects[new_project.unwrap()].clone();
+
+                let mut entry = entries[id - 1].clone();
+                entry.description = new_description.unwrap().to_string();
+                entry.project = project;
+                t.update_entry(entry).expect("API Error");
+                Ok(())
+            } else {
+                Err("Argument requirement not fulfilled".into())
+            }
+        } else {
+            Err("Could not parse values".into())
+        }
+    } else {
+        // nothing was parsed which is fine
+        Ok(())
+    }
+}
+
+fn main() {
+    let toggl = init(include_str!("../api_token")).expect("Could not connect to toggl");
+    let projects = &toggl.projects;
+    let project_ids = projects
+        .iter()
+        .map(|p| p.name.clone())
+        .collect::<Vec<String>>();
+
+    let matches = App::new("toggl")
```

Comment:
For the CLI it'd be nice to have subcommands for working with a time entry, project, workspace, etc. The most common command for me would be to start a timer, strongly preferred without a required description or project associated with it.

Reply:
Ok your usecase of empty project and empty description is now supported.
I don't think implementing sub commands is necessary at the moment as we are only handling time_entries.

Reply:
The issue is that once new resources are added, changing to subcommands will be a breaking change. Why not do so now, since the next logical feature is to add a basic way to list recent projects I'd think

Reply:
Ok I moved the time entries to the subcommand time_entry (and te for easier access).